src/text.c

```diff
@@ -273,7 +273,9 @@ ModMaskText(struct xkb_context *ctx, const struct xkb_mod_set *mods,
     if (unlikely(mask & ~((UINT64_C(1) << mods->num_mods) - 1))) {
         /* If we get a mask that cannot be expressed with the known modifiers,
          * print it as hexadecimal */
-        snprintf(buf + pos, sizeof(buf) - pos, "0x%"PRIx32, mask);
+        const int ret = snprintf(buf, sizeof(buf), "0x%"PRIx32, mask);
+        if (ret >= 0 && (size_t) ret < sizeof(buf))
```

Comment:
Would change to an assert since we are not handling failure here. If the `else` ever happens, it's no good.

Reply:
Fix with assertions